bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareRegistry.java

```diff
@@ -43,6 +43,7 @@
  * <ul>
  *
  * @author Thomas Höfer - Initial contribution
+ * @author Dimitar Ivanov - The firmwares are provided by thing and version
```

Comment:
While refactoring it anyway, could you please turn this class into an interface and move the implementation into a class named `org.eclipse.smarthome.core.thing.internal.firmware.FirmwareRegistryImpl`?